lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,654 @@
+    output_script = ""
+    script = ""
+
+    module = AnsibleModule(
+        argument_spec={
+            'device': {'required': True, 'type': 'str'},
+            'align': {
+                'default': 'optimal',
+                'choices': ['none', 'cylinder', 'minimal', 'optimal'],
+                'required': False,
+                'type': 'str'
+            },
+            'number': {'default': None, 'type': 'int', 'required': False},
+
+            # unit <unit> command
+            'unit': {
+                'default': 'KiB',
+                'choices': [
+                    's', 'B', 'KB', 'KiB', 'MB', 'MiB', 'GB', 'GiB', 'TB',
+                    'TiB', '%', 'cyl', 'chs', 'compact'
+                ],
+                'required': False,
+                'type': 'str'
+            },
+
+            # mklabel <label-type> command
+            'label': {
+                'choices': [
+                    'aix', 'amiga', 'bsd', 'dvh', 'gpt', 'loop', 'mac', 'msdos',
+                    'pc98', 'sun', ''
+                ],
+                'required': False,
+                'type': 'str'
+            },
+
+            # mkpart <part-type> [<fs-type>] <start> <end> command
+            'part_type': {
+                'default': 'primary',
+                'choices': ['primary', 'extended', 'logical'],
+                'required': False,
+                'type': 'str'
+            },
+            'part_start': {'default': '0%', 'type': 'str', 'required': False},
+            'part_end': {'default': '100%', 'type': 'str', 'required': False},
+
+            # name <partition> <name> command
+            'name': {'type': 'str', 'required': False},
+
+            # set <partition> <flag> <state> command
+            'flags': {'type': 'list', 'required': False},
+
+            # rm/mkpart command
+            'state': {
+                'choices': ['present', 'absent'],
+                'required': False,
+                'type': 'str'
+            }
+        },
+        required_together=[
+            ['number', 'state']
+        ],
+        supports_check_mode=True,
+    )
+
+    # Data extraction
+    device      = module.params['device']
+    align       = module.params['align'].lower()
+    number      = module.params['number']
+    unit        = module.params['unit']
+    label       = module.params['label']
+    part_type   = module.params['part_type'].lower()
+    part_start  = module.params['part_start']
+    part_end    = module.params['part_end']
+    name        = module.params['name']
+    state       = module.params['state']
+    flags       = module.params['flags']
+
+    # Conditioning
+    if state:
+        state = state.lower()
+    if number and number < 0:
+        module.fail_json(msg="The partition number must be non negative.")
+
+    # Read the current disk information
+    current_device = get_device_info(device, unit)
+    current_parts = current_device['partitions']
+
+    if state and state == 'present':
```

Comment:
Hmm, that is twisted IMO. I would prefer e.g. state=info over an unset state, because most modules default state to present. I think we need to proper discussion on how to tackle this, not just for this module but for all modules. A lot of modules have a _facts variant that returns information and I think we are moving slowly to a new concept of facts_plugins that would be used for returning state information.